src/components/Map/Map.tsx

```diff
@@ -1,14 +1,62 @@
 import 'leaflet/dist/leaflet.css';
 
+import { Feature, GeoJsonProperties, Geometry } from 'geojson';
+import { useEffect, useState } from 'react';
 import { MapContainer, ZoomControl } from 'react-leaflet';
 
+import NutritionChoropleth from '@/app/Nutrition/NutritionLayer';
 import { MAP_MAX_ZOOM, MAP_MIN_ZOOM } from '@/domain/constant/Map';
+import { useSidebar } from '@/domain/contexts/SidebarContext';
+import { GlobalInsight } from '@/domain/enums/GlobalInsight';
 import { MapProps } from '@/domain/props/MapProps';
+import GlobalDataRepositoryImpl from '@/infrastructure/repositories/GlobalDataRepositoryImpl';
 
 import { AlertContainer } from './Alerts/AlertContainer';
 import VectorTileLayer from './VectorTileLayer';
 
 export default function Map({ countries, disputedAreas }: MapProps) {
+  const { selectedMapType } = useSidebar();
+  const GlobalRepo = new GlobalDataRepositoryImpl();
+  const [countryStyles, setCountryStyles] = useState<{ [key: number]: L.PathOptions }>({});
+  const getFillColor = (dataType: string): string => {
+    switch (dataType) {
+      case 'actual':
+        return '#FFB74D';
+      case 'predicted':
+        return '#E3F2FD';
+      default:
+        return '#52525b';
+    }
+  };
+
+  // Fetch and preprocess data
+  useEffect(() => {
+    GlobalRepo.getNutritionData()
+      .then((global) => {
```

Comment:
we are not using promise.then, instead use await + the data should not be fetched there if that is the use case, but in the parent component of the map, which is a server component. However this data could be fetched conditionally, if the `selectedMapType = "Nutrition"`, but again in the parent component